integrationTests/consensus/consensusSigning_test.go

```diff
@@ -0,0 +1,95 @@
+package consensus
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/integrationTests"
+	"github.com/stretchr/testify/assert"
+)
+
+func initNodesWithTestSigner(
+	numNodes,
+	consensusSize,
+	numInvalid uint32,
+	roundTime uint64,
+	consensusType string,
+) []*integrationTests.TestConsensusNode {
+
+	fmt.Println("Step 1. Setup nodes...")
+
+	nodes := integrationTests.CreateNodesWithTestConsensusNode(
+		1,
+		int(numNodes),
+		int(consensusSize),
+		roundTime,
+		consensusType,
+	)
+
+	for _, nodesList := range nodes {
+		displayAndStartNodes(nodesList)
+	}
+
+	time.Sleep(p2pBootstrapDelay)
+
+	if numInvalid < numNodes {
+		for i := uint32(0); i < numInvalid; i++ {
+			iCopy := i
+			nodes[0][i].MultiSigner.CreateSignatureShareCalled = func(privateKeyBytes, message []byte) ([]byte, error) {
+				fmt.Println("invalid sig share from ",
+					getPkEncoded(nodes[0][iCopy].NodeKeys.Pk),
+				)
+				return []byte("invalid sig share"), nil
+			}
+		}
+	}
+
+	return nodes[0]
+}
+
+func TestConsensusWithInvalidSigners(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	numNodes := uint32(4)
+	consensusSize := uint32(4)
+	numInvalid := uint32(1)
+	roundTime := uint64(1000)
+	numCommBlock := uint64(8)
+
+	nodes := initNodesWithTestSigner(numNodes, consensusSize, numInvalid, roundTime, blsConsensusType)
```

Comment:
I checked the implementation of this method, and on CreateNodesWithTestConsensusNode there are nodes created for both shards and metachain.

Below in this test all nodes in every shard (including metachain) receive the same *noceForRoundMap*.
This means that if one of the shard/meta consensus is not reached the test still shows pass, as the same map is used by all nodes.
Please add a TODO: use different nonceforround maps for each node
Please create a task for that, it can come in a different PR as we already had this problem on the existing consensus test node.

Reply:
added TODOs